src/tls.c

```diff
@@ -3013,6 +3016,10 @@ static void TLSX_CSR_Free(CertificateStatusRequest* csr, void* heap)
         break;
     }
 
+#ifdef WOLFSSL_TLS13
+    if (csr->response.buffer != NULL)
+        XFREE(csr->response.buffer, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
Please add `heap`

Reply:
Added a way to use a heap hint. Also removed `responseBuffer` from internal API's that don't use it anywhere.